angularjs-portal-home/src/main/webapp/my-app/layout/widget/controllers.js

```diff
@@ -103,47 +102,85 @@ define(['angular'], function (angular) {
       }
     };
 
-    $scope.changePref = function (userPreference) {
+    $scope.cycleUnits = function (){
 
-      if (userPreference === 'C') {
-        $scope.changeToC();
-        $scope.showMetric = true;
-      } else {
-        $scope.changeToF();
+      var userPreference = $scope.nextUnits;
```

Comment:
I don't get it. In line 90 `$scope.nextUnits` was set to `C` regardless of what came back from the preferences web service, wasn't it? So isn't the rest of `populateWidgetContent()` going to always clobber the user's stored preference?